src/test/scala/org/ergoplatform/nodeView/wallet/scanning/ScanningPredicateSpecification.scala

```diff
@@ -0,0 +1,161 @@
+package org.ergoplatform.nodeView.wallet.scanning
+
+import org.ergoplatform.{ErgoBox, ErgoScriptPredef, P2PKAddress}
+import org.ergoplatform.ErgoBox.R1
+import org.ergoplatform.utils.ErgoPropertyTest
+import org.ergoplatform.utils.generators.ErgoTransactionGenerators
+import scorex.crypto.hash.Digest32
+
+import scala.util.Random
+
+class ScanningPredicateSpecification extends ErgoPropertyTest with ErgoTransactionGenerators {
+
+  val testDelay = 720 // to construct mining rewards scripts
+
+  //helper function to change random byte
+  private def mutateRandomByte(source: Array[Byte]): Array[Byte] = {
+    val sourceModified = source.clone()
+    val idx = Random.nextInt(sourceModified.length)
+    sourceModified.update(idx, ((sourceModified(idx) + 1) % Byte.MaxValue).toByte)
+    sourceModified
+  }
+
+  property("equals - p2pk") {
+    forAll(ergoAddressGen) { p2pkAddress =>
+      //look for exact p2pk script bytes
+      val box = ErgoBox(value = 1, p2pkAddress.script, creationHeight = 0)
+      val scriptBytes = p2pkAddress.script.bytes
+      EqualsScanningPredicate(R1, scriptBytes).filter(box) shouldBe true
+
+      //then change random byte
+      val scriptBytesModified = mutateRandomByte(scriptBytes)
+      EqualsScanningPredicate(R1, scriptBytesModified).filter(box) shouldBe false
+
+      //skip first byte in the filter
+      EqualsScanningPredicate(R1, scriptBytes.tail).filter(box) shouldBe false
+    }
+  }
+
+  property("equals - miner prop") {
+    forAll(proveDlogGen) { pk =>
+      val minerProp = ErgoScriptPredef.rewardOutputScript(testDelay, pk)
+      val mpBytes = minerProp.bytes
+
+      //look for exact miner script bytes
+      val box = ErgoBox(value = 1, minerProp, creationHeight = 0)
+      EqualsScanningPredicate(R1, mpBytes).filter(box) shouldBe true
+
+      //then change random byte
+      val scriptBytesModified = mutateRandomByte(mpBytes)
+      EqualsScanningPredicate(R1, scriptBytesModified).filter(box) shouldBe false
```

Comment:
This test may be unstable, because not every random byte change will falsify the predicate.

Reply:
For Equals predicate, changing an byte should falsify the check indeedi